documentation/architecture/decisions/0005-deploying-prometheus-to-private-networks-in-aws.md

```diff
@@ -0,0 +1,226 @@
+would be able to see it in their own AWS console.  However, they would
+likely not have ssh access to the instance itself.
+
+One possible issue with this model is that we're beholden on the
+client to provide us with a CIDR range to deploy onto, and depending
+on their existing estate, private IPs may be in short supply.
+
+You're also dependent on their networking arrangements.  You will need
+to ask questions like:
+
+ - how can you get in and out of the network?
+ - do you need to download packages to install?  how will that work?
+ - do you need to download security updates?  how will that work?
+
+The answers to these questions may be different for different client
+teams.  This means that our prometheus pattern needs to be flexible
+enough to cope with these differences, which will take extra
+engineering effort.
+
+We would need to work out what the integration point between our teams
+would be.  This could be:
+
+ - terraform outputs that appear in a remote state file
+   - or stereotyped resource names/tags which can be matched using
+     data sources
+
+Whether or not we go with this option for deploying prometheus, if we
+want to do ec2 service discovery (described above), prometheus will
+need some sort of IAM access into the client team account anyway.
+
+### Use VPC Peering to provide access for prometheus to scrape target infrastructure
+
+[VPC Peering][] is a feature which allows you to establish a
+networking connection between two VPCs.  In particular, this is a
+peering arrangement which means that the two networks on either side
+of the VPC Peering arrangement cannot share the same IP address
+ranges.
+
+Crucially for us, the VPCs can be in different AWS accounts.
+
+This means that we could build Prometheus in an account we own, and it
+could access client team infrastructure over the peering connection.
+Running our own infrastructure in our own account without being
+dependent on client teams providing anything to us would make for
+smoother operations and deployments for us.
+
+This has a drawback for the client in that it adds extra points of
+ingress and egress for traffic in the client networks.  This increases
+the attack surface of the network which makes doing a security audit
+harder and makes it harder to have confidence in the security of the
+system.
+
+There's also a drawback in terms of the combination of connections: as
+RE builds more services that might be provided to client teams, and as
+we extend these services to more client teams, we end up with N*M VPC
+peering connections to maintain.
+
+As RE (and techops more broadly) provides more services, client teams
+end up having to consider more VPC Peering connections in their
+security analyses, and this doesn't feel like a particularly scalable
+way for us to offer services to client teams.
+
+Finally, we believe that VPC Peering is something that has to be
+accepted manually on the receiving account console (at least when
```

Comment:
Small detail but ooking at Terraform, it does appear to be manual but could be done using Terraform as opposed to having to use the account console https://www.terraform.io/docs/providers/aws/r/vpc_peering_accepter.html